cmd/osd/main.go

```diff
@@ -483,9 +483,10 @@ func start(c *cli.Context) error {
 			Cluster:       cm,
 			StoragePolicy: sp,
 			Security: &sdk.SecurityConfig{
-				Role:           rm,
-				Tls:            tlsConfig,
-				Authenticators: authenticators,
+				Role:                         rm,
+				Tls:                          tlsConfig,
+				Authenticators:               authenticators,
+				PublicVolumeCreationDisabled: c.Bool("publicvolumecreationdisabled"),
```

Comment:
`--public-volume-create-allowed=true` and set the default to `true`, then use `!c.Bool('...')`